src/index.js

```diff
@@ -126,7 +126,13 @@ export default function prepass(
         context = assign(assign({}, context), c.getChildContext());
       }
 
-      return prepass(rendered, visitor, context);
+      let nodeToRender = rendered;
+
+      if (Array.isArray(nodeToRender)) {
+        nodeToRender = h(Fragment, null, rendered);
+      }
+
```

Comment:
Can‘t we do something along the line of `return Promise.allSettled(nodeToRender.map(node => prepass(node, visitor, context))`

Need to polyfill `Promise.allSettled` though.

Reply:
I went for `Promise.all` since when one Promise rejects we gonna bail out from traversal anyways